### PR TITLE
Update QUICKSTART with the latest context prop changes.

### PR DESCRIPTION
**Pull request by carlosalberto**

### CODE REVIEW

File: QUICKSTART.md [hunk 1]
```diff
@@ -164,8 +164,15 @@ For more details how to read context from remote processes, see
...
 ### Context Propagation
 
+In-process propagation leverages [gRPC Context](https://grpc.github.io/grpc-java/javadoc/io/grpc/Context.html),
+a well established context propagation library, contained in a small artifact, which is non-dependent on the
+entire gRPC engine.
+
 OpenTelemetry provides a text-based approach to propagate context to remote services using the
-[W3C Trace Context](https://www.w3.org/TR/trace-context/) HTTP headers. The following presents an
+[W3C Trace Context](https://www.w3.org/TR/trace-context/) HTTP headers. For doing in-process
+propagation, [gRPC Context]() is used (contained in a single, small artifact non-dependent on
```
Comment (thisthat) [inline QUICKSTART.md]:
```suggestion
[W3C Trace Context](https://www.w3.org/TR/trace-context/) HTTP headers.
```

File: QUICKSTART.md [hunk 2]
```diff
@@ -181,13 +188,17 @@ HttpTextFormat.Setter<HttpURLConnection> setter =
...
-outGoing.setAttribute("http.method", "GET");
-outGoing.setAttribute("http.url", url.toString());
-HttpURLConnection transportLayer = (HttpURLConnection) url.openConnection();
-// Inject the request with the context
-tracer.getHttpTextFormat().inject(outGoing.getContext(), transportLayer, setter);
-// Make outgoing call
+try (Scope scope = tracer.withSpan(outGoing)) {
+  // Semantic Convention
+  outGoing.setAttribute("http.method", "GET");
+  outGoing.setAttribute("http.url", url.toString());
```
Comment (arminru) [inline QUICKSTART.md]: This makes it look like if setting attributes was only allowed within the scope. You might want to move that out to make it clear this is not the case.

File: QUICKSTART.md (hunk 1 above)
Comment (carlosalberto) [reply QUICKSTART.md]: Ooops, somehow this ended up repeated. Fixed.

File: QUICKSTART.md (hunk 2 above)
Comment (carlosalberto) [reply QUICKSTART.md]: I'd rather add a note such as `setting attributes does not require the Span to be associated to the current Context`, as moving that out of the scope would also give users a bad idea (that you set the attributes first *and then* you set the `Span` active). Would that work for you?

File: QUICKSTART.md [hunk 3]
```diff
@@ -207,10 +216,11 @@ HttpTextFormat.Getter<HttpExchange> getter =
 };
 ...
 public void handle(HttpExchange he) {
-    // Extract the context from the request
-    SpanContext ctx = tracer.getHttpTextFormat().extract(he, getter);
+  // Extract the SpanContext and other elements from the request.
+  Context ctx = tracer.getHttpTextFormat().extract(Context.current(), he, getter);
```
Comment (bogdandrutu) [inline QUICKSTART.md]: `he`/`ctx` are too Go like for the Java world :) Probably we should use better variable names.

File: QUICKSTART.md [hunk 4]
```diff
@@ -207,10 +216,11 @@ HttpTextFormat.Getter<HttpExchange> getter =
 };
 ...
 public void handle(HttpExchange he) {
```
Comment (jkwatson) [inline QUICKSTART.md]: Can we put a comment up above about where this `HttpExchange` class comes from?

File: QUICKSTART.md (hunk 2 above)
Comment (arminru) [reply QUICKSTART.md]: Sounds reasonable!

File: QUICKSTART.md (hunk 3 above)
Comment (carlosalberto) [reply QUICKSTART.md]: Done.

File: QUICKSTART.md (hunk 4 above)
Comment (carlosalberto) [reply QUICKSTART.md]: I linked `HttpExchange` to https://docs.oracle.com/javase/8/docs/jre/api/net/httpserver/spec/com/sun/net/httpserver/HttpExchange.html

Does that work for you?

File: QUICKSTART.md (hunk 2 above)
Comment (carlosalberto) [reply QUICKSTART.md]: Done.

File: QUICKSTART.md (hunk 4 above)
Comment (jkwatson) [reply QUICKSTART.md]: perfect (always scary to be using stuff in `com.sun`, but it's good to at least have the connection)

**Pull request merged**